test/e2e/cross_os/test_job_submissions.py

```diff
@@ -676,3 +654,192 @@ def test_worker_uses_job_attachment_configuration(
+                if os.environ["OPERATING_SYSTEM"] == "linux"
+                else '$InputFolder = "{{Param.DataDir}}\\files"\n$OutputFile = "{{Param.DataDir}}\\output_file.txt"\n$combinedContent = ""\n$files = Get-ChildItem -Path $InputFolder -File\nforeach ($file in $files) {\n    $combinedContent += [IO.File]::ReadAllText($file.FullName)\n}\n$sha256 = [System.Security.Cryptography.SHA256]::Create().ComputeHash([System.Text.Encoding]::UTF8.GetBytes($combinedContent))\n$hashString = [System.BitConverter]::ToString($sha256).Replace("-", "").ToLower()\nSet-Content -Path $OutputFile -Value $hashString -NoNewLine'
+            )
+        ],
+    )
+    def test_worker_uses_job_attachment_sync(
+        self,
+        deadline_resources: DeadlineResources,
+        deadline_client: DeadlineClient,
+        hash_string_script: str,
+    ) -> None:
+        # Verify that the worker sync job attachment correctly and report the progress correctly as well
+
+        job_bundle_path: str = os.path.join(
+            os.path.dirname(__file__),
+            "job_attachment_bundle_large",
+        )
+        file_path: str = os.path.join(job_bundle_path, "files")
+
+        os.mkdir(job_bundle_path)
+        os.mkdir(file_path)
+
+        # Create 2500 very small files to transfer
+        for i in range(2500):
+            file_name: str = os.path.join(job_bundle_path, "files", f"file_{i+1}.txt")
+            with open(file_name, "w") as file_to_write:
+                file_to_write.write(str(i))
+
+        # Calculate the hash of all the files content combine
+        combined_string: str = ""
+        for file_name in sorted(os.listdir(file_path)):
+            file: str = os.path.join(file_path, file_name)
+            # Open the file and read its contents
+            with open(file, "r") as file_string:
+                file_contents: str = file_string.read()
+
+            # Concatenate the file contents to the combined string
+            combined_string += file_contents
+
+        combined_hash: str = hashlib.sha256(combined_string.encode()).hexdigest()
+
+        # JA template to get all files and compute the hash
+        job_parameters: List[Dict[str, str]] = [
+            {"name": "DataDir", "value": job_bundle_path},
+        ]
+        with open(os.path.join(job_bundle_path, "template.json"), "w+") as template_file:
+            template_file.write(
+                json.dumps(
+                    {
+                        "specificationVersion": "jobtemplate-2023-09",
+                        "name": "AssetsSync",
+                        "parameterDefinitions": [
+                            {
+                                "name": "DataDir",
+                                "type": "PATH",
+                                "dataFlow": "INOUT",
+                                "userInterface": {
```

Comment:
nit, not useful.

Reply:
I believe we need this to pass in the folder path to sync job attachment?

Reply:
The `userInterface` property is only useful if using a GUI to submit the job. Job attachments only uses the `dataFlow` property of a parameter.

Reply:
Yeah, what Daniel's comment suggests, we can delete the user interface block.